dragonfly/building.py

```diff
@@ -0,0 +1,241 @@
+        assert len(unique_stories) > 0, 'Building must have at least one Story.'
+        for story in unique_stories:
+            assert isinstance(story, Story), \
+                'Expected dragonfly Story. Got {}'.format(type(story))
+            story._parent = self
+        self._unique_stories = unique_stories
+
+        # process the input multipliers
+        if multipliers is None:
+            self._multipliers = tuple(1 for i in self._unique_stories)
+        else:
+            assert len(multipliers) == len(self._unique_stories), 'The length of ' \
+                'Building multipliers must match the length of unique_stories. ' \
+                '{} != {}'.format(len(multipliers), len(self._unique_stories))
+            new_mults = []
+            for m in multipliers:
+                assert m >= 1, \
+                    'Story multiplier must be greater or equal to 1. Got {}.'.format(m)
+                new_mults.append(int(m))
+            self._multipliers = multipliers
+
+        self._parent = None  # _parent will be set when Story is added to a Building
+        self._properties = BuildingProperties(self)  # properties for extensions
+
+    @property
+    def unique_stories(self):
+        """A tuple of only unique Story objects that form the Building."""
+        return self._unique_stories
+
+    @property
+    def multipliers(self):
+        """A tuple of integers denoting how many times each unique_story is repeated."""
+        return self._multipliers
+
+    @property
+    def all_stories(self):
+        """A list of all Story objects that form the Building."""
+        all_stories = []
+        for story, m in zip(self._unique_stories, self._multipliers):
+            all_stories.append(story)
+            if m != 1:
+                for i in range(m - 1):
+                    m_vec = Vector3D(0, 0, story.floor_to_floor_height * (i + 1))
+                    new_story = story.duplicate()
+                    new_story.move(m_vec)
+                    all_stories.append(new_story)
+        return all_stories
+
+    @property
+    def unique_room_polygons(self):
+        """A list of the unique RoomPolygon objects that form the Building."""
+        rooms = []
+        for room in self._unique_stories:
+            rooms.extend(room.room_polygons)
+        return rooms
+
+    @property
+    def all_room_polygons(self):
+        """A list of all RoomPolygon objects that form the Building."""
+        rooms = []
+        for room in self.all_stories:
+            rooms.extend(room.room_polygons)
+        return rooms
+
+    @property
+    def footprint_geometry(self):
+        """A ladybug_geometry Face3D object representing the footprint."""
+        raise NotImplementedError(
+            'Building footprint_geometry property has not yet been implemented.')
+        # TODO: implement footprint_geometry once we have 2D booleans in ladybug_geometry
+
+    @property
+    def context_geometry(self):
```

Comment:
I know why are you adding this here to automate simulation for each building but I feel `context_geometry` shouldn't be part of the `building` class.

Reply:
What should it be a part of then?  Or should it be a method like `get_context_representation()` instead of a property like it is here?

Reply:
I think `ContextGeometry` can be its own object which is group of geometries which will be added to a simulation as shading objects. A `building` can be located in different `contexts`. I don't think a context necessarily belongs to a building or the other way around. They are two different independent objects.

Reply:
So I think this is a really good suggestion if we end up adding a dragonfly `Model` object that parallels the honeybee `Model` object.  This way, people can add all of the `Buildings` they want to a Model and then add `ContextGeometry` objects to make a cohesive simulate-able unit.

However, you will see that this PR has no such `Model` object and I think that, at least for most local simulation workflows, we don't need a `Model` object in dragonfly.  Instead, we can do everything with the honeybee `Model` object. I am imagining that the user creates their dragonfly buildings and then connects them all up to a `to_honeybee` component. This component will accept settings for how to split up the input buildings into honeybee `Models`, whether this be one giant model, one `Building` per model with context representations of the other buildings, or maybe some finer-grained divisions down the line.

Whatever the final workflow is and regardless of whether we add `ContextGeometry` and a dragonfly `Model` object, we will need some way to easily get a simplified geometric representation of a `Building` so that it can inform shading of the other buildings around it. I would agree that it should not be a property of the `Building` and I would also agree that I picked a bad name here. For now, I changed it to be a method that accepts a `tolerance` input and I named the method `shade_representation`. I think this is a suitable solution for now and, if we find down the road that we really need a `ContextGeometry` object, we can always change it.

Reply:
Thanks! You should probably open an issue for this one and link to this conversation.